torch/csrc/api/include/torch/enum.h

```diff
@@ -0,0 +1,69 @@
+#pragma once
+
+#include <c10/util/variant.h>
+#include <torch/csrc/WindowsTorchApiMacro.h>
+
+#define TORCH_ENUM_DECLARE(name) \
+namespace torch { \
+namespace enumtype { \
+  struct name {}; \
+} \
+TORCH_API extern const enumtype::name k##name; \
+}
+
+#define TORCH_ENUM_DEFINE(name) \
+namespace torch { \
+const enumtype::name k##name; \
+}
+
+TORCH_ENUM_DECLARE(Linear)
+TORCH_ENUM_DECLARE(Conv1D)
+TORCH_ENUM_DECLARE(Conv2D)
+TORCH_ENUM_DECLARE(Conv3D)
+TORCH_ENUM_DECLARE(ConvTranspose1D)
+TORCH_ENUM_DECLARE(ConvTranspose2D)
+TORCH_ENUM_DECLARE(ConvTranspose3D)
+TORCH_ENUM_DECLARE(Sigmoid)
+TORCH_ENUM_DECLARE(Tanh)
+TORCH_ENUM_DECLARE(ReLU)
+TORCH_ENUM_DECLARE(LeakyReLU)
+TORCH_ENUM_DECLARE(FanIn)
+TORCH_ENUM_DECLARE(FanOut)
+
+namespace torch {
+
+/// Variable of `Nonlinearity` type can take one of the following values:
+/// - `torch::kLinear`
```

Comment:
@yf225 Aren't these comments redundant? the accepted values are defined in the body of declaration itself.

Reply:
I am a bit worried that people won't directly see the mapping from `enumtype::Linear` to `torch::kLinear`, and this comment should help them discover that.

Reply:
@ShahriarSS I changed the enum type from `enumtype::Linear` to `enumtype::torch::kLinear`, so that people can more easily figure out that they should use `torch::kLinear`.